napari_plugin_engine/validation.py

```diff
@@ -0,0 +1,125 @@
+import inspect
+import json
+import os
+from types import FunctionType
+from typing import Callable, Dict, List, Tuple, Union
+
+from napari.plugins import hook_specifications  # specification to be migrated
```

Comment:
yeah, probably after we are comfortable enough to freeze the specifications, otherwise developers would have to import napari to validate against the specification to be certain their plugin is working

Reply:
> have to import napari to validate against the specification to be certain their plugin is working

why is it bad to have to import napari to see if a napari plugin is working in napari?